api/plugins/session-auth.ts

```diff
@@ -0,0 +1,37 @@
+import {
+  DoneFuncWithErrOrRes,
+  FastifyPluginCallback,
+  FastifyReply,
+  FastifyRequest
+} from 'fastify';
+import fp from 'fastify-plugin';
+
+const sessionAuth: FastifyPluginCallback = (fastify, _opts, done) => {
+  fastify.decorate('authenticateSession', authenticateSession);
+
+  function authenticateSession(
+    req: FastifyRequest,
+    res: FastifyReply,
+    done: DoneFuncWithErrOrRes
+  ) {
+    if (req.session.user === undefined) {
+      res.statusCode = 401;
+      return res.send({ msg: 'Unauthorized' });
+    }
+    done();
```

Comment:
```suggestion
      void res.send({ msg: 'Unauthorized' });
    } else {
    done();
    }
```
Since we're only calling this function for its side-effects, it's better not to return anything. It implies that the return value might be used somehow, but that's not the case.